lib/api/src/externals/function.rs

```diff
@@ -621,11 +621,39 @@ mod inner {
         };
     }
 
+    macro_rules! from_to_native_wasm_type_same_size {
+        ( $( $type:ty => $native_type:ty ),* ) => {
+            $(
+                #[allow(clippy::use_self)]
+                unsafe impl FromToNativeWasmType for $type {
+                    type Native = $native_type;
+
+                    #[inline]
+                    fn from_native(native: Self::Native) -> Self {
+                        unsafe {
+                            std::mem::transmute::<$native_type, $type>(native)
+                        }
+                    }
+
+                    #[inline]
+                    fn to_native(self) -> Self::Native {
+                        unsafe {
+                            std::mem::transmute::<$type, $native_type>(self)
+                        }
+                    }
+                }
+            )*
+        };
+    }
+
     from_to_native_wasm_type!(
         i8 => i32,
         u8 => i32,
         i16 => i32,
-        u16 => i32,
+        u16 => i32
+    );
+
+    from_to_native_wasm_type_same_size!(
```

Comment:
I really don't understand the naming here, `from_to_X_Y`? I read it expecting "from X to Y"? Maybe "define casts" instead?

Reply:
🤷 That's what the thing is called right now, it's a trait for bidirectional conversion

Reply:
OK. I still dislike the naming but it doesn't need to rolled into this PR.

Reply:
Perhaps file an issue for it, I remember Ivan talking about it at the time. I thought the name was a bit strange too, but it seemed like a good option given the other names he considered